unitylibs/scripts/utils.js

```diff
@@ -14,29 +14,167 @@ export const [setLibs, getLibs] = (() => {
+  ];
+})();
+
+export function decorateArea(area = document) {}
+
+const miloLibs = setLibs('/libs');
+
+const { createTag, getConfig, loadStyle } = await import(`${miloLibs}/utils/utils.js`);
+export { createTag, loadStyle, getConfig };
+const { decorateDefaultLinkAnalytics } = await import(`${miloLibs}/martech/attributes.js`);
+export { decorateDefaultLinkAnalytics };
+
+export function getGuestAccessToken() {
+  try {
+    const { token } = window.adobeIMS.getAccessToken();
+    return `Bearer ${token}`;
+  } catch (e) {
+    return '';
+  }
+}
+
+export function getHeaders(apiKey) {
+  return {
+    'Content-Type': 'application/json',
+    Authorization: getGuestAccessToken(),
+    'x-api-key': apiKey,
   };
+}
 
-  (async function loadLCPImage() {
-    const marquee = document.querySelector('.marquee');
-    if (!marquee) {
-      eagerLoad(document, 'img');
-      return;
+export function defineDeviceByScreenSize() {
+  const DESKTOP_SIZE = 1200;
+  const MOBILE_SIZE = 600;
+  const screenWidth = window.innerWidth;
+  if (screenWidth >= DESKTOP_SIZE) return 'DESKTOP';
+  if (screenWidth <= MOBILE_SIZE) return 'MOBILE';
+  return 'TABLET';
+}
+
+export async function loadSvg(src) {
+  try {
+    const res = await fetch(src, { mode: 'no-cors' });
+    if (!res.status === 200) return null;
+    const svg = await res.text();
+    return svg;
+  } catch (e) {
+    return '';
+  }
+}
+
+export function loadImg(img) {
+  return new Promise((res) => {
+    img.loading = 'eager';
+    img.fetchpriority = 'high';
+    if (img.complete) res();
+    else {
+      img.onload = () => res();
+      img.onerror = () => res();
     }
-  
-    // First image of first row
-    eagerLoad(marquee, 'div:first-child img');
-    // Last image of last column of last row
-    eagerLoad(marquee, 'div:last-child > div:last-child img');
-  }());
+  });
+}
+
+export async function createActionBtn(btnCfg, btnClass, iconAsImg = false, swapOrder = false) {
+  const txt = btnCfg.innerText;
+  const img = btnCfg.querySelector('img[src*=".svg"]');
+  const actionBtn = createTag('a', { href: '#', class: `unity-action-btn ${btnClass}` });
+  if (img) {
+    let btnImg = null;
+    const { pathname } = new URL(img.src);
+    const libSrcPath = `${getUnityLibs().split('/unitylibs')[0]}${pathname}`;
+    if (iconAsImg) btnImg = createTag('img', { src: libSrcPath });
+    else btnImg = await loadSvg(libSrcPath);
+    const btnIcon = createTag('div', { class: 'btn-icon' }, btnImg);
+    actionBtn.append(btnIcon);
+  }
+  if (txt) {
+    const btnTxt = createTag('div', { class: 'btn-text' }, txt.split('\n')[0].trim());
+    if (swapOrder) actionBtn.prepend(btnTxt);
+    else actionBtn.append(btnTxt);
+  }
+  return actionBtn;
 }
 
-export async function useMiloSample() {
-  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+async function createErrorToast() {
+  const alertImg = await fetch(`${getUnityLibs()}/img/icons/alert.svg`).then((res) => res.text());
```

Comment:
Both the SVGs can be loaded in parallel.

Reply:
Fixed